generic/tests/nfs_corrupt.py

```diff
@@ -71,6 +74,20 @@ def setup(self, force_start=False):
 
         :param force_start: Whether to make NFS service start anyway.
         """
+        error.context("Checking available space to export")
+        tmpdir = os.path.split(self.nfs_dir)[0]
+        out = utils.system_output("df %s" % tmpdir,
+                                  ignore_status=True).split()[-3]
+
+        available_size = float(utils_misc.normalize_data_size(out,
+                                                              order_magnitude="M"))
+        required_size = float(utils_misc.normalize_data_size(self.required_size,
+                                                             order_magnitude="M"))
+        logging.debug("Space available: %fM, space needed: %fM"
+                      % (available_size, required_size))
+        if available_size < required_size:
+            raise exceptions.TestError("Not enough free space to setup NFS")
```

Comment:
you can assign test to self.test in __init__ on line 22:
```python
     def __init__(self, test, params, ip="localhost"):
         self.test = test
```
So you can use `self.test.error("Not enough free space to setup NFS")`.
And you may want to update all the raise clause using `from autotest.client.shared import error` as it will not be supported for python 3.